drivers/hmis/app/graphql/types/hmis_schema/ce_referral.rb

```diff
@@ -43,9 +43,10 @@ def self.authorized?(object, ctx)
     summary_field :custom_status, HmisSchema::CeCustomReferralStatus, null: true
     summary_field :client_id, ID, null: false
     summary_field :client_name, String, null: true, description: 'The name of the referred client. Always available to those who can view the full referral, even without full client record access.'
-    # Special case: Client is a "summary field" because it doesn't require referral visibility, but resolving it does require permission to view the client record.
     summary_field :created_at, GraphQL::Types::ISO8601DateTime, null: false
     summary_field :source_enrollment_id, ID, null: false
+    # source_project_name is resolved separately from source_enrollment as a summary field to minimize data exposure
+    summary_field :source_project_name, String, null: false
```

Comment:
I'm proposing to add `source_project_name` as a new summary-level field, even though it is also resolvable through the `sourceEnrollment` field. As we currently have it set up, `sourceEnrollment` is not a summary-level field, so it's only resolvable by those who can view the referral. I thought about elevating it to a summary field, but it contains a lot of data (like assessments and household members) that are not needed to solve the current requirement, so I've gone with this approach, which creates less churn. Let me know what you think!

Reply:
https://github.com/greenriver/hmis-warehouse/pull/5999/commits/3fa5e401f11c1111006e2046b2dfd10138409c50